Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -917,8 +960,7 @@ def network_stop(self, removestate, removecontent):
                 else:
                     self.set_vod_mode(False)
                     self.handle.pause()
-                    pstate.set('state', 'engineresumedata', self.handle.save_resume_data()
-                               if isinstance(self.tdef, TorrentDef) else None)
+                    pstate.set('state', 'engineresumedata', defer.maybeDeferred(self.save_resume_data))
```

Comment:
when it's not available, it will stored as deferred. 
But when actually saving it, it will get data first, then save it.